### PR TITLE
fix console.timeLog

### PR DESCRIPTION
**Pull request by lqqyt2423**
fix console.timeLog data arguments aren't logged in output

https://github.com/oven-sh/bun/issues/3821

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
Thank you for this 

Are the arguments supposed to be stringified? What happens if you pass a symbol? Where are the tests?

**Comment by lqqyt2423** [comment]
@Jarred-Sumner I changed code, and add some test.

File: test/js/web/console/console-timeLog.test.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { expect, it } from "bun:test";
+import { bunExe } from "harness";
+
+it("should log to console correctly", async () => {
+  const { stderr, exited } = spawn({
+    cmd: [bunExe(), import.meta.dir + "/console-timeLog.js"],
+    stdin: null,
+    stdout: "pipe",
+    stderr: "pipe",
+    env: {
```
Comment (Electroid) [inline test/js/web/console/console-timeLog.test.ts]:
```suggestion
    env: bunEnv,
```

File: test/js/web/console/console-timeLog.test.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { file, spawn } from "bun";
+import { expect, it } from "bun:test";
+import { bunExe } from "harness";
```
Comment (Electroid) [inline test/js/web/console/console-timeLog.test.ts]:
```suggestion
import { bunExe, bunEnv } from "harness";
```

**Review by Electroid — CHANGES_REQUESTED** [review]

File: src/bun.js/bindings/exports.zig [hunk 3]
```diff
@@ -3284,13 +3285,33 @@ pub const ZigConsoleClient = struct {
...
+            .remaining_values = &[_]JSValue{},
+            .globalThis = global,
+            .ordered_properties = false,
+            .quote_strings = false,
+        };
+        var console = global.bunVM().console;
+        var writer = console.error_writer.writer();
+        const Writer = @TypeOf(writer);
+        for (0..args_len) |i| {
+            const arg = args[i];
```
Comment (paperclover) [inline src/bun.js/bindings/exports.zig]:
```suggestion
        for (args[0..args_len]) |arg| {
```

**Review by paperclover — APPROVED** [review]

**Comment by paperclover** [comment]
this test failure on linux x64 might be existing/flaky but it did trip:
 - `test/js/third_party/prisma/prisma.test.ts`
the code for this looks fine and is unrelated to the error i think.

File: test/js/web/console/console-timeLog.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+console.timeLog("label", "a %s b", "c", "d");
+console.timeLog("label", 0, -0, 123, -123, 123.567, -123.567, Infinity, -Infinity);
+console.timeLog("label", true, false);
+console.timeLog("label", null, undefined);
+console.timeLog("label", Symbol("Symbol Description"));
+console.timeLog("label", new Date(Math.pow(2, 34) * 56));
+console.timeLog("label", [123, 456, 789]);
+console.timeLog("label", { name: "foo" });
+console.timeLog("label", { a: 123, b: 456, c: 789 });
+console.timeLog("label");
```
Comment (RadhiRasho) [inline test/js/web/console/console-timeLog.js]:
```suggestion
console.timeLog("label");
console.timeEnd("labal");
```

Don't forget the end

**Review by paperclover — APPROVED** [review]

**Comment by paperclover** [comment]
timeLog tests fail. i believe this is simply a mistake in the test itself?
image

**Comment by lqqyt2423** [comment]
@paperdave fixed

**Review by paperclover — APPROVED** [review]

**Comment by paperclover** [comment]
thanks!

**Pull request merged**